### PR TITLE
Snap changes

### LINKED ISSUE #130: Investigate Updating to Latest Snap API
In [January of 2021](https://medium.com/metamask/breaking-changes-to-the-metamask-provider-are-here-7b11c9388be9), MetaMask made a number of breaking changes to the Snap provider API. The largest change was to remove the injected `window.web3` instance. Additionally, the following breaking changes were made

- Stop 0-padding `eth_chainId` return values for some chains
- Stop emitting `chainIdChanged`, and instead emit `chainChanged`
- Removing `ethereum.publicConfigStore`
- Removing the `ethereum._metamask.isEnabled` experimental method
- Removing the `ethereum._metamask.isApproved` experimental method

This ticket is for tracking an investigation into how these API changes will affect the project in this repository. See [MetaMask's migration guide](https://docs.metamask.io/guide/provider-migration.html) for more details.

### LINKED ISSUE #137: [Bug] Snap Doesn't Load Network Details
After connecting the Snap, the UI loads but the network details do not load. In the console, I see the following error:
`MetaMask - RPC Error: Cannot read properties of undefined (reading 'slice')`

Edit: I am using v10.16.2-flask.0 of MetaMask Flask.

Possibly related to #136 (?)

image

### PR DESCRIPTION
**Pull request by irubido**
### TODO

- [x] network change
- [x] block number
- [x] hash
- [x] address
- [x] public key
- [x] balance
- [x] export private key
- [x] transfer
- [x] sign message

closes #130
closes #137

### CODE REVIEW

File: packages/snap/src/polkadot/account.ts [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
  * @param wallet
  */
 export async function getKeyPair(wallet: Wallet): Promise<KeyringPair> {
-  // get app key and wait for api to be ready
-  const [appKey] = await Promise.all([wallet.getAppKey(), cryptoWaitReady()]);
+
+  const bip44Node = (await wallet.request({
+    method: `snap_getBip44Entropy_${kusamaCoinType}`,
+    params: [],
+  })) as JsonBIP44CoinTypeNode;
```
Comment (mpetrunic) [inline packages/snap/src/polkadot/account.ts:20]: is this derivation for new metamask or old one?

**Review by mpetrunic — COMMENTED** [review]
Two things:
- stop stubbing wasm crypto module (this should work in snap now)
- apply chainsafe eslint configuration

File: packages/snap/src/polkadot/account.ts (hunk 1 above)
Comment (BeroBurny) [reply packages/snap/src/polkadot/account.ts:20]: It is a new one, I don't see a point in supporting older versions.

**Comment by BeroBurny** [comment]
> Two things:
> 
> * stop stubbing wasm crypto module (this should work in snap now)
> * apply chainsafe eslint configuration

this pr si kinda big and this is good to have it more in smaller steps for improving the quality there is issue #139
so we should process by merging this one?

**Comment by BeroBurny** [comment]
> do you wanna address that any problem here or in future PR?

in future one

**Pull request merged**